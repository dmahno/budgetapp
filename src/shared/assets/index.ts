export type TImageKeys = '1.jpg' | '2.jpg' | '3.jpg' | '4.jpg' | '5.jpg';

import image1 from './images/1.jpg';
import image2 from './images/2.jpg';
import image3 from './images/3.jpg';
import image4 from './images/4.jpg';
import image5 from './images/5.jpg';

export const images: Record<TImageKeys, string> = {
  '1.jpg': image1,
  '2.jpg': image2,
  '3.jpg': image3,
  '4.jpg': image4,
  '5.jpg': image5,
};
