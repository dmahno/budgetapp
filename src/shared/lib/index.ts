export {formatNumber} from './formatNumber/formatNumber';
export {parseNumber} from './parseNumber/parseNumber';
