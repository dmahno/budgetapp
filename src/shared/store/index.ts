export * from './AuthStore/AuthStore';
export * from './ThemeStore/ThemeStore';
