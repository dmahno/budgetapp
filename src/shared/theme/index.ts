export {configDarkTheme} from './tokens/configDarkTheme';
export {configLightTheme} from './tokens/configLightTheme';
