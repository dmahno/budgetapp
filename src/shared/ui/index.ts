export {Logo} from './Logo/Logo';
export {ReactRouterCustomLink} from './ReactRouterCustomLink/ReactRouterCustomLink';
export {DynamicIcon} from './DynamicIcon/DynamicIcon';
export {TapButton} from './TapButton/TapButton';
