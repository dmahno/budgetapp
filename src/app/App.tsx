import {Suspense} from 'react';

import {Spinner} from 'shared/ui';

import {AppRouter} from './Providers/AppRouter';

export function App() {
  return (
    <Suspense fallback={<Spinner />}>
      <AppRouter data-testid="app-router" />
    </Suspense>
  );
}
