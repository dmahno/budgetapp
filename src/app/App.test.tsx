import {describe, it, expect, vi} from 'vitest';
import {Suspense} from 'react';

import {render, screen} from 'shared/tests/test-utils';
import {Spinner} from 'shared/ui';

import {App} from './App';

vi.mock('./Providers/AppRouter', () => ({
  AppRouter: () => <div data-testid="app-router">App Router Content</div>,
}));

describe('App', () => {
  it('renders AppRouter within Suspense', () => {
    render(<App />);
    const appRouterElement = screen.getByTestId('app-router');
    expect(appRouterElement).toBeInTheDocument();
    expect(appRouterElement).toHaveTextContent('App Router Content');
  });

  it('renders Suspense fallback while loading', async () => {
    const SuspendedComponent = () => {
      throw Promise.resolve();
    };

    render(
      <Suspense fallback={<Spinner />}>
        <SuspendedComponent />
      </Suspense>,
    );

    const fallbackElement = screen.getByTestId('spinner');
    expect(fallbackElement).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const {asFragment} = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
