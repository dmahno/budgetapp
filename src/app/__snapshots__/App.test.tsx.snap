// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="app-router"
  >
    App Router Content
  </div>
</DocumentFragment>
`;
